c++/src/Writer.cc

```diff
@@ -122,9 +127,17 @@ namespace orc {
   }
 
   WriterOptions& WriterOptions::setFileVersion(const FileVersion& version) {
-    // Only Hive_0_11 version is supported currently
-    if (version.getMajor() == 0 && version.getMinor() == 11) {
+    // Only Hive_0_11 and Hive_0_12 version are supported currently
+    if (version.getMajor() == 0 && (version.getMinor() == 11 || version.getMinor() == 12)) {
```

Comment:
My suggestion is to use this logic to implement `WriterOptions::getRleVersion()`.

Reply:
Done